src/main.js

```diff
@@ -0,0 +1,186 @@
+// SPDX-License-Identifier: Apache-2.0
+
+import { resolve } from "path";
+import { downloadTool, extractTar, extractZip } from '@actions/tool-cache';
+import { addPath, setFailed, getInput } from '@actions/core';
+import { platform as osPlatform, arch as osArch } from 'node:os';
+import { existsSync } from 'node:fs';
+import { execSync } from 'node:child_process';
+
+
+async function getLatestVersion() {
+    // Note: The GitHub API is rate limited to 60 requests per hour for unauthenticated users.
+    // I am not quite sure that it is reliable to query the latest version from
+    // the GitHub API because of the rate limits.
+    let version;
+    let response = await fetch(`https://api.github.com/repos/ispc/ispc/releases/latest`);
+    if(response.status !== 200) {
+        let errorDetails = await response.text();
+        console.log(`Unable to query latest version: ${response.statusText} - ${errorDetails}`);
+
+        try {
+            // Fallback to local git repository
+            const repoPath = 'ispc-repo';
+            if (!existsSync(repoPath)) {
+                execSync(`git clone --depth=1 https://github.com/ispc/ispc.git ${repoPath}`);
+            }
+
+            console.log('Fetching latest tags from the repository...');
+            execSync(`git -C ${repoPath} fetch --tags`);
+
+            let latestTagSHA = execSync(`git -C ${repoPath} rev-list --tags --max-count=1`).toString().trim();
+            version = execSync(`git -C ${repoPath} describe --tags ${latestTagSHA}`).toString().trim();
```

Comment:
Should we remove the repo after determining the version?

Reply:
Fixed